Amethyst/Model/Application.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  Application.swift
+//  Amethyst
+//
+//  Created by Ian Ynda-Hummel on 3/12/19.
+//  Copyright © 2019 Ian Ynda-Hummel. All rights reserved.
+//
+
+import Foundation
+import Silica
+
+protocol ApplicationType: Equatable {
+    associatedtype Window: WindowType
+
+    init(runningApplication: NSRunningApplication)
+
+    func title() -> String?
+    func windows() -> [AnyWindow<Window>]
+    func pid() -> pid_t
+    func windowWithTitleShouldFloat(_ windowTitle: String) -> Bool
+    func dropWindowsCache()
+    func observe(notification: String, handler: @escaping SIAXNotificationHandler) -> Bool
+    func observe(notification: String, window: AnyWindow<Window>, handler: @escaping SIAXNotificationHandler) -> Bool
+    func unobserve(notification: String)
+    func unobserve(notification: String, window: AnyWindow<Window>)
+}
+
+class AnyApplication<Application: ApplicationType>: ApplicationType {
```

Comment:
This appears to be a wrapper class. Comment here should explain why we do that (my guess: testability)